### PR TITLE
[Arm64] Fix assembly errors

### PR DESCRIPTION
**Pull request by sdmaclea**

### CODE REVIEW

**Comment by sdmaclea** [comment]
@dotnet/arm64-contrib @rahku Fixes #10971.  PTAL.

File: src/vm/arm64/asmhelpers.S [hunk 1]
```diff
@@ -1220,52 +1220,52 @@ NESTED_END StubDispatchFixupStub, _TEXT
 // ------------------------------------------------------------------
 // void* JIT_GetSharedNonGCStaticBase(SIZE_T moduleDomainID, DWORD dwClassDomainID)
 
-    LEAF_ENTRY JIT_GetSharedNonGCStaticBase_SingleAppDomain
+LEAF_ENTRY JIT_GetSharedNonGCStaticBase_SingleAppDomain, _TEXT
     // If class is not initialized, bail to C++ helper
     add x2, x0, #DomainLocalModule__m_pDataBlob
-    ldrb w2, [x2, w1]
```
Comment (sdmaclea) [inline src/vm/arm64/asmhelpers.S]: This is an invalid instruction form.  Given that both arguments are 64 bit, I assumed the following form was what was intended.

**Comment by rahku** [comment]
LGTM Thanks!

**Comment by rahku** [comment]
@dotnet-bot test Windows_NT arm64 checked

**Comment by sdmaclea** [comment]
@rahku Looks like Windows arm64 assembler does not recognize UXTW.  I am going to remove the change to Windows arm64

**Comment by sdmaclea** [comment]
@rahku Done

**Pull request merged**